File: service-notification-java/src/main/java/com/acme/services/notification/java/Worker.java
package com.acme.services.notification.java;

import io.infinitic.workers.InfiniticWorker;

public class Worker {
    public static void main(String[] args) {
        // infinitic.yml resource is from the dependant "contracts" module
        try (InfiniticWorker worker = InfiniticWorker.fromConfigResource("/infinitic.yml", "/notification.yml")) {
            worker.start();
        }
    }
}

File: service-notification-kotlin/src/main/kotlin/com/acme/services/notification/kotlin/Worker.kt
package com.acme.services.notification.kotlin

import io.infinitic.workers.InfiniticWorker

fun main() {
    // infinitic.yml resource is from the dependant "contracts" module
    InfiniticWorker.fromConfigResource("/infinitic.yml", "/notification.yml").use { worker ->
        worker.start()
    }
}
